flaml/model.py

```diff
@@ -405,32 +409,75 @@ def search_space(cls, data_size, task, **params):
 
         return search_space_dict
 
-    def _init_hf_args(self, automl_fit_kwargs: dict = None):
-        from .nlp.utils import HFArgs
-
-        hf_args = HFArgs(task=self._task)
-        fit_kwargs = automl_fit_kwargs.get("hf_args", None)
-        if fit_kwargs:
-            for key, val in fit_kwargs.items():
-                assert (
-                    key in hf_args.__dict__
-                ), "The specified key {} is not in the argument list of flaml.nlp.utils::HFArgs".format(
-                    key
-                )
-                setattr(hf_args, key, val)
-        self.hf_args = hf_args
-
-    def _update_hf_args(self, automl_pred_kwargs: dict = None):
-        if automl_pred_kwargs:
-            hf_args = automl_pred_kwargs.get("hf_args")
-            if hf_args:
-                for key, val in hf_args.items():
-                    assert (
-                        key in self.hf_args.__dict__
-                    ), "The specified key {} is not in the argument list of flaml.nlp.utils::HFArgs".format(
-                        key
-                    )
-                    setattr(self.hf_args, key, val)
+    @property
+    def checkpoint_freq(self):
+        return (
+            int(
+                min(self._training_args.num_train_epochs, 1)
+                * len(self._X_train)
+                / self._training_args.per_device_train_batch_size
+                / self._training_args.ckpt_per_epoch
+            )
+            + 1
+        )
+
+    @property
+    def fp16(self):
+        return (
+            self._kwargs.get("gpu_per_trial")
+            and self._kwargs.get("gpu_per_trial") > 0
+            and self._training_args.fp16
+        )
+
+    @property
+    def no_cuda(self):
+        return not (
+            self._kwargs.get("gpu_per_trial")
+            and float(self._kwargs.get("gpu_per_trial")) > 0
+        )
+
+    def _set_training_args(self, **kwargs):
+        from .nlp.utils import date_str, Counter
+
+        for (key, val) in kwargs.items():
+            assert key not in self.params, (
+                "Since {} is in the search space, it cannot exist in 'custom_fit_kwargs' at the same time."
+                "If you need to fix the value of {} to {}, the only way is to add a single-value domain in the search "
+                "space by adding:\n '{}': {{ 'domain': {} }} to 'custom_hp'. Please see FLAML/test/automl/test_custom_hp.py "
+                "for an example.".format(key, key, val, key, val)
+            )
+
+        """
+            If use has specified any custom args for TrainingArguments, update these arguments
+        """
+        self._training_args = self._TrainingArguments(**kwargs)
+
+        """
+            Update the attributes in TrainingArguments with self.params values
+        """
+        for key, val in self.params.items():
+            setattr(self._training_args, key, val)

```

Comment:
are all the params in self.params attributes of self._training_args? I don't think so.